tests/terraform/graph/variable_rendering/test_render_scenario.py

```diff
@@ -169,21 +172,24 @@ def match_blocks(self, expected_block_val, different_expected, got_block_type_li
         return False
 
     def match_resources(self, expected_block_val, different_expected, got_block_type_list, expected_block_name):
+        found = False
         for got_block_dict in got_block_type_list:
             for got_block_name, got_block_val in got_block_dict.items():
                 if got_block_name == expected_block_name:
-                    expected_resource_name = list(expected_block_val.keys())[0]
+                    # expected_resource_name = list(expected_block_val.keys())[0]
                     got_resource_name = list(got_block_val.keys())[0]
-                    if expected_resource_name != got_resource_name:
+                    if got_resource_name not in expected_block_val:
```

Comment:
FYI, dear reviewers: These changes are to support evaluating more than one resource of the same type (as in my test main.tf below).